test/support/test_fixed_resource_policy_parallel_utils.h

```diff
@@ -0,0 +1,201 @@
+// -*- C++ -*-
+//===---------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_TEST_FIXED_RESOURCE_PARALLEL_UTILS_H
+#define _ONEDPL_TEST_FIXED_RESOURCE_PARALLEL_UTILS_H
+
+#include "support/test_config.h"
+#include "oneapi/dpl/dynamic_selection"
+#include "support/utils.h"
+#include <unordered_map>
+#include <thread>
+
+template <bool call_select_before_submit, typename Policy, typename UniverseContainer, typename UniverseMapping>
+int
+test_submit_and_wait(UniverseContainer u, UniverseMapping map, std::vector<int> actual, int offset=0)
+{
+    using my_policy_t = Policy;
+    my_policy_t p(u, offset);
+    std::vector<int> result(u.size(), 0);
+    const int N = 5;
+
+    std::vector<std::thread> threads;
+    std::mutex m;
```

Comment:
Which code exactly we want to synchronize with this mutex?

Reply:
Unnecessary. Fixed.